src/account/operations/transaction/high_level/burning_melting/destroy_foundry.rs

```diff
@@ -0,0 +1,226 @@
+        TransactionOptions,
+    },
+    Error,
+};
+
+impl AccountHandle {
+    /// Function to destroy a foundry output with a circulating supply of 0.
+    /// Native tokens in the foundry (minted by other foundries) will be transactioned to the controlling alias
+    pub async fn destroy_foundry(
+        &self,
+        foundry_id: FoundryId,
+        options: Option<TransactionOptions>,
+    ) -> crate::Result<TransactionResult> {
+        log::debug!("[TRANSACTION] destroy_foundry");
+
+        let alias_id = *foundry_id.alias_address().alias_id();
+        let (existing_alias_output_data, existing_foundry_output_data) =
+            self.find_alias_and_foundry_output_data(alias_id, foundry_id).await?;
+
+        validate_empty_state(&existing_foundry_output_data.output)?;
+
+        let custom_inputs = vec![
+            existing_alias_output_data.output_id,
+            existing_foundry_output_data.output_id,
+        ];
+
+        let options = match options {
+            Some(mut options) => {
+                options.custom_inputs.replace(custom_inputs);
+                Some(options)
+            }
+            None => Some(TransactionOptions {
+                custom_inputs: Some(custom_inputs),
+                ..Default::default()
+            }),
+        };
+
+        let outputs = match existing_alias_output_data.output {
+            Output::Alias(alias_output) => {
+                let amount = existing_alias_output_data.amount + existing_foundry_output_data.amount;
+                let mut native_tokens_builder = NativeTokensBuilder::from(alias_output.native_tokens().clone());
+                // Transfer native tokens from foundry to alias
+                if let Output::Foundry(foundry_output) = existing_foundry_output_data.output {
+                    native_tokens_builder.add_native_tokens(foundry_output.native_tokens().clone())?;
+                } else {
+                    unreachable!("We already checked output is a foundry");
+                }
+                // Create the new alias output with updated amount, state_index and native token if not burning foundry
+                // tokens
+                let alias_output = AliasOutputBuilder::from(&alias_output)
+                    .with_alias_id(alias_id)
+                    .with_amount(amount)?
+                    .with_native_tokens(native_tokens_builder.finish()?)
+                    .with_state_index(alias_output.state_index() + 1)
+                    .finish()?;
+
+                vec![Output::Alias(alias_output)]
+            }
+            _ => unreachable!("We checked if it's an alias output before"),
+        };
+
+        self.send(outputs, options).await
+    }
+
+    /// Destroy all foundries in the given set `foundry_ids`
+    // TODO: allow destroying of multiple foundries of the same alias, currently only one foundry output per alias can
```

Comment:
Is this something that needs to be done within this PR? Or can this be done at a later point in time?
Maybe then have a issue to not forget

Reply:
I think it's fine to do it at a later point, because one can call the method to destroy a foundry alone before calling this
Also it would delay it more and in many cases it will no be possible to destroy foundry outputs at all if the native token amount didn't get melted before
But yes, creating an issue is good